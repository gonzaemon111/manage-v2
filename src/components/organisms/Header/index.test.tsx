import { composeStories } from "@storybook/react";
import { render } from "@testing-library/react";
import * as Stories from "./index.stories";

describe("Organisms/Header コンポーネント", () => {
  test("適切なclassNameが適用されている", () => {
    const { Default } = composeStories(Stories);
    const { getByText, container } = render(<Default />);
    expect(getByText("Manage")).toBeTruthy();
  });

  describe("スナップショットテスト", () => {
    const testCases = Object.values(composeStories(Stories)).map((Story) => [
      Story.storyName,
      Story,
    ]);

    test.each(testCases)("Renders %s story", async (storyName, Story) => {
      if (!Story) {
        throw new Error("Story not Found");
      }

      const tree = render(<Story />);
      expect(tree).toMatchSnapshot();
    });
  });
});
