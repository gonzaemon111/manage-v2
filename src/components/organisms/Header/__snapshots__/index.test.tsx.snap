// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/Header コンポーネント スナップショットテスト Renders Default story 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="from-blue-500 to-primary bg-gradient-to-b shadow-lg text-slate-100"
      >
        <div
          class="px-8 mx-auto max-w-7xl"
        >
          <div
            class="flex items-center justify-between h-[64px]"
          >
            <div
              class=" flex items-center"
            >
              <a
                class="flex-shrink-0 flex items-center gap-2 md:gap-4"
                href="/"
              >
                <img
                  alt="Workflow"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/_next/image?url=%2Ficon.png&w=64&q=75"
                  srcset="/_next/image?url=%2Ficon.png&w=32&q=75 1x, /_next/image?url=%2Ficon.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="30"
                />
                <span
                  class="text-slate-50 font-semibold text-xl"
                >
                  Manage
                </span>
              </a>
              <div
                class="hidden md:block"
              >
                <div
                  class="flex ml-10 space-x-4 items-center"
                >
                  <a
                    class="hover:text-slate-50 hover:underline hover:font-semibold px-3 py-2 rounded-md text-sm font-medium"
                    href="https://github.com/gonzaemon111"
                    target="_blank"
                  >
                    Github
                  </a>
                  <a
                    class="hover:text-slate-50 hover:underline hover:font-semibold px-3 py-2 rounded-md text-sm font-medium"
                    href="https://qiita.com"
                    target="_blank"
                  >
                    Qiita
                  </a>
                  <a
                    class="hover:text-slate-50 hover:underline hover:font-semibold px-3 py-2 rounded-md text-sm font-medium"
                    href="https://zenn.dev"
                    target="_blank"
                  >
                    Zenn
                  </a>
                </div>
              </div>
            </div>
            <div
              class="block"
            >
              <div
                class="flex items-center ml-4 md:ml-6"
              />
            </div>
            <div
              class="flex -mr-2 md:hidden"
            >
              <button
                class="inline-flex items-center justify-center p-2 rounded-md focus:outline-none"
              >
                <svg
                  class="w-7 h-7"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 1792 1792"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1664 1344v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zm0-512v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zm0-512v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="from-blue-500 to-primary bg-gradient-to-b shadow-lg text-slate-100"
    >
      <div
        class="px-8 mx-auto max-w-7xl"
      >
        <div
          class="flex items-center justify-between h-[64px]"
        >
          <div
            class=" flex items-center"
          >
            <a
              class="flex-shrink-0 flex items-center gap-2 md:gap-4"
              href="/"
            >
              <img
                alt="Workflow"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/_next/image?url=%2Ficon.png&w=64&q=75"
                srcset="/_next/image?url=%2Ficon.png&w=32&q=75 1x, /_next/image?url=%2Ficon.png&w=64&q=75 2x"
                style="color: transparent;"
                width="30"
              />
              <span
                class="text-slate-50 font-semibold text-xl"
              >
                Manage
              </span>
            </a>
            <div
              class="hidden md:block"
            >
              <div
                class="flex ml-10 space-x-4 items-center"
              >
                <a
                  class="hover:text-slate-50 hover:underline hover:font-semibold px-3 py-2 rounded-md text-sm font-medium"
                  href="https://github.com/gonzaemon111"
                  target="_blank"
                >
                  Github
                </a>
                <a
                  class="hover:text-slate-50 hover:underline hover:font-semibold px-3 py-2 rounded-md text-sm font-medium"
                  href="https://qiita.com"
                  target="_blank"
                >
                  Qiita
                </a>
                <a
                  class="hover:text-slate-50 hover:underline hover:font-semibold px-3 py-2 rounded-md text-sm font-medium"
                  href="https://zenn.dev"
                  target="_blank"
                >
                  Zenn
                </a>
              </div>
            </div>
          </div>
          <div
            class="block"
          >
            <div
              class="flex items-center ml-4 md:ml-6"
            />
          </div>
          <div
            class="flex -mr-2 md:hidden"
          >
            <button
              class="inline-flex items-center justify-center p-2 rounded-md focus:outline-none"
            >
              <svg
                class="w-7 h-7"
                fill="currentColor"
                height="16"
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1664 1344v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zm0-512v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zm0-512v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
